lib/Basic/LangOptions.cpp

```diff
@@ -85,28 +143,21 @@ bool contains(const StringRef (&Array)[N], const StringRef &V,
     if (distance == minDistance)
       suggestions.emplace_back(candidate);
   }
+  suggestedKind = suggestedPlatformConditionKind(Kind, V, suggestions);
   return false;
 }
 
 bool LangOptions::
 checkPlatformConditionSupported(PlatformConditionKind Kind, StringRef Value,
-                                std::vector<StringRef> &suggestions) {
+                                PlatformConditionKind &suggestedKind,
+                                std::vector<StringRef> &suggestedValues) {
```

Comment:
Would it make sense to return the suggestions as a `std::vector<std::pair<PlatformConditionKind, StringRef>> &` instead? That way we could suggest more flexible corrections as conditions warrant.

Reply:
I don't think **[(PlatformConditionKind, StringRef)]** would fit well here. Sometimes, we want multiple value suggestions for the same PlatformConditionKind (for e.g. `#if os(chaos)` should suggest multiple closest-edit-distance-matching values for the "os" kind), and sometimes, we want no value suggestions for a PlatformConditionKind (for e.g. `#if os(simulator)` should suggest only a kind change, not a value change).

Are you saying that maybe in the future, we might want to return multiple PlatformConditionKind suggestions? In that case, maybe **[(PlatformConditionKind, [StringRef])]** would be more appropriate, because we can have 0..<n value suggestions for a given kind. Nevertheless, I don't think that change should be made now -- it should rather be done when we do start returning multiple PlatformConditionKind suggestions. What do you think?

Reply:
That's fair.